### PR TITLE
#169 - Extracted Manifests from Repo

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #169 
Extracted `Manifests` from `Repo` and `AstoManifests` from `AstoRepo` as-is.

### CODE REVIEW

File: src/main/java/com/artipie/docker/Manifests.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+     * @param content Manifest content.
+     * @return Added manifest.
+     */
+    CompletionStage<Manifest> put(ManifestRef ref, Content content);
+
+    /**
+     * Get manifest by reference.
+     *
+     * @param ref Manifest reference
+     * @return Manifest found, empty if manifest is absent.
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/docker/Manifests.java:53]: @olegmoz seems like `if` is missing in the first part of the sentence

File: src/main/java/com/artipie/docker/asto/AstoManifests.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                final CompletionStage<Optional<Digest>> stage;
+                if (exists) {
+                    stage = this.asto.value(key)
+                        .thenCompose(
+                            pub -> new ByteBufPublisher(pub).asciiString()
+                        )
+                        .<Digest>thenApply(Digest.FromString::new)
+                        .thenApply(Optional::of);
+                } else {
+                    stage = CompletableFuture.completedFuture(Optional.empty());
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/docker/asto/AstoManifests.java:198]: @olegmoz I'd suggest to extract method from `CompletableFuture.completedFuture(Optional.empty())` as it's repeated several times here.

Comment (olegmoz) [reply src/main/java/com/artipie/docker/asto/AstoManifests.java:198]: @olenagerasimova sorry, I'd prefer not to. This looks like an overkill to me for quite simple code without any logic in it

File: src/main/java/com/artipie/docker/Manifests.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/docker/Manifests.java:53]: @olenagerasimova thanks, I've reworded the comment

**Pull request merged**